### DISCUSSION TITLE
Show me how you use Note Toolbar!

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#4] [Show and tell]
I'd love to see [examples](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Examples) of how you use Note Toolbar, and any plugins you may have used in the process!

Have any examples that use scripting (Dataview, Javascript, Templater)? Share it in the [Scripting discussion](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/categories/scripting).

Don't have a GitHub account? [Fill out this Google form instead](https://docs.google.com/forms/d/e/1FAIpQLSeVWHVnookJr8HVQywk5TwupU-p7vkRkSt83Q5jscR6VwpZEQ/viewform?usp=sf_link).

**Comment by omeraloni** [comment]
Periodic Notes navigation bar:
<img width="699" alt="image" src="https://github.com/chrisgurney/obsidian-note-toolbar/assets/146173/9be08f8c-6d6f-4254-8c04-152faeec09de">

made using toolbar callout, since the static menu does not have templating support.

**Reply by chrisgurney** [reply]
@omeraloni Thank you for sharing!

I'm curious, are those values (2024, Q2, April, W16) available in the Properties, or your folder structure?

It wouldn't take too much work to support the substitution of Properties (or the note title) into variables in menu labels (as it works today [for URIs](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Creating-toolbars#variables)). However, this will become non-trivial when those values don't come from a consistent location.

Open to ideas on templating!

**Reply by omeraloni** [reply]
folder structure.
I have a templater script that generates it for daily, weekly, monthly, etc.

That's how the daily callout is generated:
```javascript
    let prev = moment(date).subtract(1, offset).format(fmt);
    let next = moment(date).add(1, offset).format(fmt);
    let year = date.format('YYYY');
    let quarter = date.format('YYYY-[Q]Q');
    let month = date.format('YYYY-MMMM');
    let week = date.format('YYYY-[W]ww');

    if (key == 'Daily') {

        str = `> [!note-toolbar|border-even-sticky]\n`
        str += `> - [[${root}/${key}/${moment(prev, fmt).format('YYYY/MM-MMMM')}/${prev}|❮❮]]\n`
        str += `> - [[${root}/Yearly/${year}|${year}]]\n`
        str += `> - [[${root}/Quarterly/${year}/${quarter}|${date.format('[Q]Q')}]]\n`
        str += `> - [[${root}/Monthly/${year}/${month}|${date.format('MMMM')}]]\n`
        str += `> - [[${root}/Weekly/${year}/${week}|${date.format('[W]ww')}]]\n`
        str += `> - [[${root}/${key}/${moment(next, fmt).format('YYYY/MM-MMMM')}/${next}|❯❯]]`
    }
```

I've also created a non-callout (static?) version of the navigation bar that call commands ([Journal](https://github.com/srg-kostyrko/obsidian-journal) plugin) instead:
<img width="668" alt="image" src="https://github.com/chrisgurney/obsidian-note-toolbar/assets/146173/d1fa2c82-2f4e-4555-9e8a-c961f945b08a">

Currently I use the callout version as I like seeing the actual dates.

**Reply by Moyf** [reply]
Very interesting method, thanks for sharing!

And I got a kind-of tricky way to make it more "Dynamic" with Dataview plugin..
(Maybe too weird lol)

(1) with DV's inline script support,
![image](https://github.com/user-attachments/assets/39f73af4-ce7a-427e-a036-641fe923f78e)

we can get make a link like this:
```md
`$="[[" + moment().format("[root/]YYYY") + "|" +moment().format("YYYY")+"]]"`
```

It would be rendered as the actual year (2024), and would link to the actual note (you can change the note link's code)


(2) And if we put this code into Callout..
![image](https://github.com/user-attachments/assets/b8324bb1-f947-499e-8afc-9265e99df44a)

walla!
![image](https://github.com/user-attachments/assets/e1019bdf-44a7-4e37-9bd1-09c010c3c551)

We can make it "dynamic" - no matter where you put this callout, it would automatically show the real-time links.

___

It might be a bit redundant, but I think it's quite fun. 😄

**Comment by kmaustral** [comment]
Please keep examples coming.

**Comment by MrComputron** [comment]
I'll add my own for basic navigation:
![image](https://github.com/user-attachments/assets/738506fe-6395-472a-8cb1-a4faa27836a8)
![image](https://github.com/user-attachments/assets/a60815d8-2309-4a98-b563-e26a82087bef)
![image](https://github.com/user-attachments/assets/c36c35ec-0c62-4469-b692-9704468668c0)
![image](https://github.com/user-attachments/assets/eeb018e5-fa4a-4916-aa17-7521863b6ae8)

This little axe closes all other tabs, which is particularly useful for me as on mobile using that command takes both too many steps and instead of closing all other tabs it only closes half at a time.
![image](https://github.com/user-attachments/assets/b810d782-5158-4006-9aa1-9ad2dc80b54a)

The little sofa is to switch themes...because it's decor.

**Reply by chrisgurney** [reply]
Thanks for sharing @MrComputron !

I'm assuming you're using a theme? If so, which one? The fact that the text and the icons don't vertically align bothers me. 😅

(If I need a font please let me know which one.)

**Reply by MrComputron** [reply]
Theme is Chime, styled "Macaroon" in style settings.
Do you mean the alignment of the button icon to the drop down icons? If so I never noticed until now and now I can't unsee...

**Reply by davidszerman** [reply]
> I'll add my own for basic navigation: ![image](https://private-user-images.githubusercontent.com/46870771/360792545-738506fe-6395-472a-8cb1-a4faa27836a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4Nzg0MjEsIm5iZiI6MTc1Mzg3ODEyMSwicGF0aCI6Ii80Njg3MDc3MS8zNjA3OTI1NDUtNzM4NTA2ZmUtNjM5NS00NzJhLThjYjEtYTRmYWEyNzgzNmE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDEyMjIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0OTc0YWFkNGIyODc5NTdhZDNkOTE2MGVkN2RiZTQxYzUyZGVkNTc4NWRhMmMyMDU2MjZiMTNjN2E0MjExZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.E8BeAAMt5mg32C2l2VWsBZZaW5aLFFDspQvqDSnazpQ) ![image](https://private-user-images.githubusercontent.com/46870771/360792601-a60815d8-2309-4a98-b563-e26a82087bef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4Nzg0MjEsIm5iZiI6MTc1Mzg3ODEyMSwicGF0aCI6Ii80Njg3MDc3MS8zNjA3OTI2MDEtYTYwODE1ZDgtMjMwOS00YTk4LWI1NjMtZTI2YTgyMDg3YmVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDEyMjIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczZTBhNTZkMjg0YTQ0MDliY2E5OThmMGY5ZTFiY2E0MzEyNmUxY2QyNzA5Y2VjNjNjYWUyMGI1ZGI1NmJmZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bZ3QVKPC2A0qz2Co41rOq8G8Zs0hFNtdi6kallTpuqA) ![image](https://private-user-images.githubusercontent.com/46870771/360792693-c36c35ec-0c62-4469-b692-9704468668c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4Nzg0MjEsIm5iZiI6MTc1Mzg3ODEyMSwicGF0aCI6Ii80Njg3MDc3MS8zNjA3OTI2OTMtYzM2YzM1ZWMtMGM2Mi00NDY5LWI2OTItOTcwNDQ2ODY2OGMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDEyMjIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyYjA0MjYyMDhkNTViMTc0YjU2ODJmYTRmZTZhYjZkMTM0YWIzNGZhZDY5OGM2MzExYjQyNGJlZTAyZDIwOGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.p4I9t9X3xVakH8tQdoHo8NnWMNuUEG2WFhKui0A063s) ![image](https://private-user-images.githubusercontent.com/46870771/360792762-eeb018e5-fa4a-4916-aa17-7521863b6ae8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4Nzg0MjEsIm5iZiI6MTc1Mzg3ODEyMSwicGF0aCI6Ii80Njg3MDc3MS8zNjA3OTI3NjItZWViMDE4ZTUtZmE0YS00OTE2LWFhMTctNzUyMTg2M2I2YWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDEyMjIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2ZGFiYmIyZTk2MTIxZGIxYTQ0MTQxNTU4MDBiMDQ0ODE3YjU1ZWZiOGZkZDkwZDY1ZThmN2ZmNmZlYTEzYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2tpnwONm-igPml9dZdpmkmNjTgHMjt9HuWyuSDltac8)
> 
> This little axe closes all other tabs, which is particularly useful for me as on mobile using that command takes both too many steps and instead of closing all other tabs it only closes half at a time. ![image](https://private-user-images.githubusercontent.com/46870771/360792813-b810d782-5158-4006-9aa1-9ad2dc80b54a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4Nzg0MjEsIm5iZiI6MTc1Mzg3ODEyMSwicGF0aCI6Ii80Njg3MDc3MS8zNjA3OTI4MTMtYjgxMGQ3ODItNTE1OC00MDA2LTlhYTEtOWFkMmRjODBiNTRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzMwVDEyMjIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwYWQ2ZTFiOTkyNmQwNmM3NzYyOWFjMGU4NzI2NTdlZTRiOGE1MTc2MWFiMzdiMGVkNGE2ZjM2ZGYyOWEwZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iyoV3Xe2cHIlmTmLxiutM1a2qrcglfyVEtzHktEVXCM)
> 
> The little sofa is to switch themes...because it's decor.

Hi

How did you define File from heading ?

Thanks

**Comment by MrComputron** [comment]
I assume that there is some default Obsidian UI thing that makes button pop ups and drop downs  slide up from the bottom of the screen on mobile. 
Do you know if there is a way to change how pop ups behave? The issue with the drop downs on mobile is that if you're hitting buttons at the top of your screen, you have to move your fingers all the way to the bottom to continue in the drop down.
This makes it harder to use the drop down menus one handedly as you have to awkwardly juggle your phone past the first touch. Would be nice if possible to have drop downs behave more as they do on desktop.

**Reply by chrisgurney** [reply]
Another user had that request over in #109 

[As far as I could tell](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/109#discussioncomment-10149805), I couldn't find a way to override Obsidian's menu UI, at least with CSS alone.

As mentioned, I'd be curious if anyone could find plugins, themes, or CSS snippets that achieves this change, so I'd have a starting point to look into options.

**Reply by MrComputron** [reply]
Yeah exactly that. If it **is** possible it would be a huge improvement for ease of use on mobile, especially for larger phones.
I'm curious now, do you know how this is handled on folding devices?

**Reply by chrisgurney** [reply]
@MrComputron I don't have a folding device, but I've seen mention of more than one user using one. I believe Obsidian is using the `.is-phone` CSS class in those cases, which triggers the bottom-style menu.

A floating button is a good option for phones as it's near the bottom of the screen (by default) and thus the jump to the menu is not as big a distance for the thumb to travel. Have you given that a try?

In #78 for example the floating button wasn't as far down the screen as that user (with a folding phone) wanted. However, with the Style Settings plugin you can now adjust the vertical position, to put the button anywhere you like. Take a look at `Style Settings > Note Toolbar > Floating Button > Positioning > Y position on mobile (in vh units)`

**Reply by MrComputron** [reply]
I hadn't really considered the floating button as I don't usually use it. I tried it out and it's definitely good for when you need one handed use. One issue i have is that when I'm using the the floating button, I would expect certain buttons to be closer to the bottom if they are more critical. When it is a toolbar at the top to be used with two hands it's not really a consideration but for one handed mobile use the less movement your thumb makes the better the experience.

I'm not sure if there is an easy UI solution to this or even a practical solution within the obsidian framework. One partial customization could be an option to flip the order of the floating button items, so that if you're reading left to right on a standard note Toolbar, a floating button would have the option to read top to bottom with a starting point that is the initial thumb placement, if that makes sense.

Another thing that would make it easier to read is - if possible - reducing the width of the pop up menu so that the text of the items is closer to the thumb position. I've attached a very detailed mockup of what this might look like.

![Screenshot_20240823_145100_Obsidian](https://github.com/user-attachments/assets/3cf9af28-6e4d-4cb1-9b00-7a5e13f0969e)

**Reply by MrComputron** [reply]
Mostly because there is a lot of negative space by the thumb making it harder to easily and accurately target the intended button.

**Reply by chrisgurney** [reply]
I'll continue to look into whether I can style that menu at all.

What if I can get the icons to right-align down the right edge of the menu, while leaving the labels on the left?

Reversing the order is also an interesting idea.

By the way, for that icon without a label, if you add a tooltip it will show up in the menu.

**Reply by MrComputron** [reply]
I think right-aligning the icons is a good start. Maybe worth trying with right align icon and right align text to its left?

I tried several variations of label vs tooltip and and was unable to make the tooltip show when there was no label. But your comment gave me an idea:

Since on mobile the tools are stacked vertically and each has a lot more room horizontally,  it could be interesting to display the tooltip in the pop up and the label on the toolbar to make use of that horizontal space.

**Reply by chrisgurney** [reply]
Hmm that's odd. Can you share a screenshot of the settings for that item?

Here's one of mine:

<img src="https://github.com/user-attachments/assets/d2062b09-500f-4b92-b366-b5f30011445e" width="400"/>

And the menu:

<img src="https://github.com/user-attachments/assets/10a3dc85-ea3c-4e48-8078-9b37b89e580d" width="400"/>

**Reply by chrisgurney** [reply]
@MrComputron Did you try/manage to get your icon without a label to display text (in the menu)?

**Comment by Moyf** [comment]
![image](https://github.com/user-attachments/assets/181312cb-1798-4895-91a9-4ad2d98891de)

For my daily note, I would put those buttons:
- **Today**: Jump to today's daily note
  - rather than using the command "Open daily note", I wrote a TP script to automatically decide: 
    1) Switch to the tab if it's already open;
    2) Open in new tab
    3) Create the daily not if it's not exists
- **Daily Home**: Go to my "Daily" homepage where I can view more information
- **Morning Journal**: Go to today's journal note
- **Cur Week**: Go to this week's weekly note
- **Cur Month**: Same
- `---`
- **Re-order list**: A TP Script to re-order selected list
- **Common Functions**: My another toolbar, which contains the shortcuts I usually use
  - **Open today's daily note** - repeated with "Today" but I would need it elsewhere ; )
  - **Work Doc Note**: A TP Script to open the newest created note in my "WorkDoc" Folder
  - **Open in VSCode**: Call the command of "Open In VSCode" Plugin
  - **Repeat Command**: Repeat the last command

![image](https://github.com/user-attachments/assets/56805ace-b2f9-4787-91cb-762896717a58)

BTW, I would also use the "Common Functions" as my **Default floating button** for all those pages without NTB config:
![image](https://github.com/user-attachments/assets/802e7020-824b-404e-8b62-b9d6d6104c29)

> Actually I would like it to be displayed in all notes ;)

**Reply by momo-in-github** [reply]
请问怎么做出来这个命令的？「重复命令 ：重复最后一个命令」

**Reply by Moyf** [reply]
> 请问怎么做出来这个命令的？「重复命令 ：重复最后一个命令」

有个单独的插件 There's a plugin to do it:
https://github.com/3C0D/obsidian-repeat-last-commands

**Comment by Moyf** [comment]
Oh, and today I saw this image in one of my chat group:
![8b8e6109cf9e94a20f8ccde08da92308](https://github.com/user-attachments/assets/e5a48618-a8b1-4ce8-8655-05368e3c753f)

It appears that someone is using NTB as a "Buttons" plugin, allowing a single note to contain multiple buttons (With no content but a fixed-top NTB!) - which can then be moved into the sidebar. 

![image](https://github.com/user-attachments/assets/82f70ff1-15ef-4b18-b12d-e13a24d895eb)
I utilized the Buttons plugin to achieve this, but I was also surprised by this method of usage. 😄

**Comment by chrisgurney** [comment]
Now that you can share toolbars [in v1.14](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/153), I'd love to see more examples of how you use Note Toolbar.

I'll be adding links to the wiki so installing toolbars is just a click away!

**Reply by brucetm** [reply]
This is a great idea. I'd love to be able to browse other people's toolbar creations, pick one I like, and customize it. Instead of starting from scratch.
